hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -370,6 +377,28 @@ public synchronized HoodieActiveTimeline reloadActiveTimeline() {
     return activeTimeline;
   }
 
+  /**
+   * Returns next instant time in the correct format. Lock is enabled by default.
+   */
+  public String createNewInstantTime() {
+    return createNewInstantTime(true);
+  }
+
+  /**
+   * Returns next instant time in the correct format.
+   *
+   * @param shouldLock whether the lock should be enabled to get the instant time.
+   */
+  public String createNewInstantTime(boolean shouldLock) {
```

Comment:
I am thinking if we can simplify all the `shouldLock` passing somehow. Would it be possible to just have the lock configuration passed down and let the `TimeGenerator` decide? Default can be inprocess lock provider?

Reply:
My worry is all the places we pass `true` `false` and how it may make it hard to maintain .

Reply:
We've been thinking through this for a while and we want to avoid the dead-lock, the lock providers other than in-process is kind of not reentrant now.